hydra-node/exe/hydra-node/Main.hs

```diff
@@ -26,11 +26,18 @@ import Hydra.Node (
   createHydraHead,
   runHydraNode,
  )
+import Hydra.Node.Version (gitRevision, showFullVersion, version)
 import Hydra.Option (Option (..), parseHydraOptions)
 
 main :: IO ()
-main = do
-  Option{nodeId, verbosity, host, port, peers, apiHost, apiPort, monitoringPort} <- identifyNode <$> parseHydraOptions
+main =
+  identifyNode <$> parseHydraOptions >>= \case
+    option
+      | displayVersion option == True -> putStr (showFullVersion version gitRevision)
+      | otherwise -> startHydraNode option
```

Comment:
Instead of parsing an `option` into a `Bool`, displaying version is typically done (what I have seen so far) using [`infoOption`](https://hackage.haskell.org/package/optparse-applicative-0.16.1.0/docs/Options-Applicative.html#v:infoOption) and have the parser exit `main` directly by showing the version.

Reply:
Also, we could (should?) use a library instead of our own template haskell splices. There is [gitrev](https://hackage.haskell.org/package/gitrev-1.3.1).

Both things combined would make the `--version` parser (which would be applied like the `helper`) something like:
```
versionOption =
  infoOption (showVersion version <> "-" <> $(gitHash))
             (long "version" <> help "Show version")
```
This way we alternatively could also use `gitDescribe` to not use the cabal version at all (debatable).

Reply:
But maybe I'm missing something and that `file-embed` is also important to us? Maybe @KtorZ knows when that was useful in the wallet?

Reply:
IIUC the file-embed is useful for the case when the binary is built in an environment without access to git. This can be the case in CI depending on how it is done, I have seen and done that in the past.
I have never used `infoOption`. If I understand how it works, the version is output to `stderr` because it's handled as a `ParseError` which is something I find really annoying. The `--version` option is useful not only for humans consumption but also for other pieces of software that can use this information as a sanity check before, eg. deployment. That's the main reason why I always do it that way, and I was not surprised to see it done that way in the wallet too.

Reply:
Note that we could also use `git describe` in the TH and simplify `Version` to use only that as the version.
`gitrev` seems indeed more robust than what we do, I will update the code to use that instead.

Reply:
My bad: using `infoOption`actually outputs the version on stdout, I am going to update the PR to use that.

Reply:
About the version parsing, on cardano-node (& ogmios, which shamelessly stole from it) they ended up doing things slightly differently, so that both flags and commands would be accepted; because some prefer calling `exe version` instead of `exe --version` etc.. 

https://github.com/KtorZ/cardano-ogmios/blob/master/server/src/Ogmios/App/Options.hs#L201-L215

About the file-embed, that is indeed one work-around for building within docker or nix for instance.